pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -72,6 +73,7 @@ const (
 	NodeTypeRetry     NodeType = "Retry"
 	NodeTypeSkipped   NodeType = "Skipped"
 	NodeTypeSuspend   NodeType = "Suspend"
+	NodeTypeAgent     NodeType = "Agent"
```

Comment:
this does not make sense to me, no node can be agent? they are `NodeTypeHTTP` surely?